x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mockConfig, mockLogger } from '../__mocks__';
+
+import { createEnterpriseSearchRequestHandler } from './enterprise_search_request_handler';
+
+jest.mock('node-fetch');
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mock;
+const { Response } = jest.requireActual('node-fetch');
+
+const responseMock = {
+  ok: jest.fn(),
+  notFound: jest.fn(),
+  customError: jest.fn(),
+};
+const KibanaAuthHeader = 'Basic 123';
+
+interface IMockResponse {
+  results: object[];
+  meta?: {
+    page?: {
+      total_results: number;
+    };
+  };
+}
```

Comment:
Hmm, I'm not too sure we need a custom interface for a test file. Can we just do:

```
hasValidData: (body?: any)
```

below instead?

Reply:
Ah, you're right. Not necessary. I felt like I was getting dinged by TS for using `any` other places in this project, but it seems to be OK here. Are there different rules between test vs app code?

Reply:
Accepted your suggestion to make this change, and then removed the old interface here: https://github.com/elastic/kibana/pull/75487/commits/9f742e1955d3720f9c014f1d87aba279753963f8

Reply:
Yes! We allow `any` in tests but not in non-test-files (to make our lives easier when mocking during tests). I set that rule up in the plugin MVP PR here:

https://github.com/elastic/kibana/blob/9cafade2b9d4a22a364c921e94a30c54e82e2ee0/.eslintrc.js#L1129-L1139